plugins/ckan_operators/package_operator.py

```diff
@@ -212,4 +213,86 @@ def execute(self, context):
             if self.lists_match( self.package_a, self.package_b ):
                 return {"result": True}
 
-        # TODO: add logic for comparing only packages and not lists of packages
\ No newline at end of file
+        # TODO: add logic for comparing only packages and not lists of packages
+
+class GetCkanPackageListOperator(BaseOperator):
+    """
+    Get full package list
+    """
+    @apply_defaults
+    def __init__(
+        self,
+        address: str,
+        apikey: str,
+        **kwargs
+    ) -> None:
+        super().__init__(**kwargs)
+        self.ckan = ckanapi.RemoteCKAN(apikey=apikey, address=address)
+
+
+    def execute(self, context):
+        package_list = self.ckan.action.package_list()
+        logging.info(f'package list {package_list}')
+        return package_list
+
+class CheckCkanResourceDescriptionOperator(BaseOperator):
+    """
+    Check if the description of a resource is missing
+    """
+    @apply_defaults
+    def __init__(
+        self,
+        input_dag_id: str = None,
+        base_url : str = None,
+        resource_url : str = None,
+        **kwargs
+    ) -> None:
+        super().__init__(**kwargs)
+        self.input_dag_id = input_dag_id
+        self.base_url = base_url
+        self.resource_url = resource_url
+
+    def execute(self, context):
+        ti = context['ti']
+        package_list = ti.xcom_pull(task_ids = self.input_dag_id)
+        result_info = {}
+        for package_name in package_list:
+            url = self.base_url + package_name
+            resources = requests.get(url).json()['result']['resources']
+            for resource in resources:
+                if str(resource["datastore_active"]) == "True":
+                    url = self.resource_url + resource["id"]
+                    fields = requests.get(url).json()['result']['fields']
+
+                    field_flag = [True if ('info' in item.keys()) and (item['info']['notes'] is not None) else False for item in fields]
+                    resource_flag = all([flag is False for flag in field_flag])
+
+                    if resource_flag:
+                        logging.warning(f'Resource description MISSING! package id or name: {package_name} resource id: {resource["id"]}')
+                        
+                        if package_name in result_info:
+                            result_info[package_name] = result_info[package_name] + ', ' + resource["id"]
+                        else:
+                            result_info[package_name] = resource["id"]
+
+                    else:
+                        logging.info('Resource description OK!')
+        return {"package_and_resource": result_info}
+
+class PrintCkanResponseOperator(BaseOperator):
```

Comment:
Since you dont use this in your DAG, you can delete it :)

Reply:
right!!